src/fp-paradigm/js/createStore.js

```diff
@@ -0,0 +1,38 @@
+const createStore = (reducer, initialState = []) => {
+    let state = initialState;
+
+    return {
+        dispatch: (action) => {
+            state = reducer(state, action);
+
+            // Could be better if we subscribe to state change
+            rerender(state);
+        },
+        getState: () => state,
+    };
+};
+
+const reducer = (state, action) => {
```

Comment:
Ye I like everything here, but I don't know how is it realized in redux itself, think that there if you initialize a reducer with an empty state, it will throw an error. I think that creating a variable inside a function goes against FP. Instead maybe each time in a reducer initialize the state, ex: const reducer = (state = blabla, action) ...; I checked; If you don't mind I will throw you some code, and you can look at it, and the result it gives:
const createStore = (reducer, initialState = []) => {
    let state = initialState;

    return {
        dispatch: (action) => {
            state = reducer(state, action);

            // Could be better if we subscribe to state change
            console.log(state);
        },
        getState: () => state,
    };
};
const reducer = (state = [{id: 1, task: "todo", completed: false}], action) => {
    switch(action.type) {
        case 'ADD_TODO':
            return [
                ...state,
                {
                    id: state.length,
                    task: action.task,
                    completed: false,
                },
            ];
        case 'TOGGLE_TODO':
            return state.map((todo) => {
                if (todo.id === action.id) {
                    todo.completed = !todo.completed;
                }
                return todo;
            });
        case 'DELETE_TODO':
            return state.filter((_, i) => action.id !== i);
    }
};
const { getState, dispatch } = createStore(reducer);
dispatch({type: "ADD_TODO", task: "blabla"});

Reply:
`const createStore = (reducer, initialState = []) => {
    let state = initialState;

    return {
        dispatch: (action) => {
            state = reducer(state, action);

            // Could be better if we subscribe to state change
            console.log(state);
        },
        getState: () => state,
    };
};
const reducer = (state = [{id: 1, task: "todo", completed: false}], action) => {
    switch(action.type) {
        case 'ADD_TODO':
            return [
                ...state,
                {
                    id: state.length,
                    task: action.task,
                    completed: false,
                },
            ];
        case 'TOGGLE_TODO':
            return state.map((todo) => {
                if (todo.id === action.id) {
                    todo.completed = !todo.completed;
                }
                return todo;
            });
        case 'DELETE_TODO':
            return state.filter((_, i) => action.id !== i);
    }
};
const { getState, dispatch } = createStore(reducer);
dispatch({type: "ADD_TODO", task: "blabla"});`

Reply:
I think, without impurity, we can manipulate our state. Dan's idea is, reducers should be pure. The same goal we've achieved in saga and other libraries we can make async request and get the result, after dispatch our result. What about `combineReducers` and `combineState` it is different topic, and i will write / finish them till evening. The idea of dispatch is simple, same as for mobx. Just to rerender specific part of our UI, makes them great, which we can't for now, need to learn philosophy of React / Angular, how do they do this. For now, we just making a clone of this things so, no worries.

Reply:
interested in seing combineState